json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -207,6 +210,12 @@ public function update_settings() {
 						$updated[ $key ] = $value;
 					};
 					break;
+				case 'jetpack_protect_whitelist':
+					$result = jetpack_protect_save_whitelist( $value, false );
```

Comment:
Can we do some sort of check if $value is really an array here before saving the data. 
This might need to be implemented in jetpack_protect_save_whitelist function.